src/main/java/org/bahmni/sms/web/security/OpenMRSAuthenticator.java

```diff
@@ -0,0 +1,54 @@
+package org.bahmni.sms.web.security;
+
+import org.apache.http.HttpResponseFactory;
+import org.apache.http.impl.DefaultHttpResponseFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bahmni.sms.SMSProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class OpenMRSAuthenticator {
+
+    private static final Logger logger = LogManager.getLogger(OpenMRSAuthenticator.class);
+    private static final String WHOAMI_URL = "/bahmnicore/whoami";
+    public static final String OPENMRS_SESSION_ID_COOKIE_NAME = "JSESSIONID";
+
+    @Autowired
+    private SMSProperties properties;
+
+    public ResponseEntity authenticate(String sessionId) {
```

Comment:
same question here. Would have preferred authenticate returning simple status .. rather than a response entity .. rest can be taken care at controller level.